src/components/ApiRefRouting/ScrollRouter.js

```diff
@@ -92,10 +102,30 @@ export const ScrollRouter = ({ children }) => {
     }),
     [stopTrackingElement, trackElement, onLinkClick, isScrollingDown],
   );
+  const sideNavContextValue = React.useMemo(
+    () => ({
+      // Make these no-op, cuz we're already tracking elements. This would be
+      // used by TrackedContent, but we'll already have Route components reporting
+      stopTrackingElement: () => {},
+      trackElement: () => {},
+      activeContent: {
+        ref: activeNode,
+        id: routeMap.get(activeNode),
+      },
+    }),
+    [activeNode],
+  );
 
-  return <Context.Provider value={contextValue}>{children}</Context.Provider>;
+  return (
+    <Context.Provider value={contextValue}>
+      <SideNavProgressContext.Provider value={sideNavContextValue}>
```

Comment:
This is a little hacky, but short of a full refactor of sidenav + scrollrouter to get them to share some of the underlying code, I couldn't come up with something better. These have very similar behaviors, they just track different elements and respond slightly differently.